gcloud/monitoring/timeseries.py

```diff
@@ -0,0 +1,157 @@
+        adds ``"resource_type"``.
+        """
+        # pylint: disable=attribute-defined-outside-init
+        try:
+            return self._labels
+        except AttributeError:
+            labels = {'resource_type': self.resource.type}
+            labels.update(self.resource.labels)
+            labels.update(self.metric.labels)
+            self._labels = labels
+            return self._labels
+
+    def header(self, points=None):
+        """Copy everything but the point data.
+
+        :type points: list of :class:`Point`, or None
+        :param points: An optional point list.
+
+        :rtype: :class:`TimeSeries`
+        :returns: The new time series object.
+        """
+        points = list(points) if points else []
+        return self._replace(points=points)
+
+    @classmethod
+    def _from_dict(cls, info):
+        """Construct a time series from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`TimeSeries`
+        :returns: A time series object.
+        """
+        metric = Metric._from_dict(info['metric'])
+        resource = Resource._from_dict(info['resource'])
+        metric_kind = info['metricKind']
+        value_type = info['valueType']
+        points = [Point._from_dict(p) for p in info.get('points', [])]
+        return cls(metric, resource, metric_kind, value_type, points)
+
+    def __repr__(self):
+        """Return a representation string with the points elided."""
+        return (
+            '<TimeSeries with {num} points:\n'
+            ' metric={metric!r},\n'
+            ' resource={resource!r},\n'
+            ' metric_kind={kind!r}, value_type={type!r}>'
+        ).format(
+            num=len(self.points),
+            metric=self.metric,
+            resource=self.resource,
+            kind=self.metric_kind,
+            type=self.value_type,
+        )
+
+
+class Point(collections.namedtuple('Point', 'end_time start_time value')):
+    """A single point in a time series.
+
+    This is a named tuple. See :func:`collections.namedtuple`.
+
+    :type end_time: string
+    :param end_time: The end time in RFC3339 UTC "Zulu" format.
+
+    :type start_time: string or None
+    :param start_time: An optional start time in RFC3339 UTC "Zulu" format.
+
+    :param value: The metric value. This can be a scalar or a distribution.
+    """
+    __slots__ = ()
+
+    @classmethod
+    def _from_dict(cls, info):
+        """Construct a Point from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`Point`
+        :returns: A point object.
+        """
+        end_time = info['interval']['endTime']
+        start_time = info['interval'].get('startTime')
+        value_type, value = next(six.iteritems(info['value']))
```

Comment:
Actually, secretly there is an assertion that you only expect one item, so you could use `tuple`-unpacking to make that assertion and assign to your values:


```python
(value_type, value), = info['value'].items()
```